src/lib/orders.ts

```diff
@@ -349,20 +356,83 @@ export async function getOrders(props: {
 export async function submitOrderCancellationByIdAction(
   orderId: string,
 ): Promise<never> {
+  try {
+    await submitOrderCancellationById(orderId);
+    console.log(`Cancellation for Order ${orderId} submitted.`);
+    process.exit(0);
+  } catch (e) {
+    if (e instanceof Error) {
+      return logAndQuit(e.message);
+    } else {
+      return logAndQuit("An unknown error occurred");
+    }
+  }
+}
+
+async function cancelAllOpenOrdersAction(options: { force?: boolean }) {
   const loggedIn = await isLoggedIn();
   if (!loggedIn) {
     logLoginMessageAndQuit();
   }
 
+  // Fetch all open orders
+  const openOrders = await getOrders({ only_open: true });
+
+  if (openOrders.length === 0) {
+    console.log("No open orders to cancel.");
+    process.exit(0);
+  }
+
+  console.log(`Found ${openOrders.length} open order(s).`);
+
+  // Confirm cancellation unless force option is used
+  if (!options.force) {
+    const shouldProceed = await confirm({
+      message: "Are you sure you want to cancel all open orders?",
+      default: false,
+    });
+
+    if (!shouldProceed) {
+      console.log("Operation cancelled.");
+      process.exit(0);
+    }
+  }
+
+  // Cancel all open orders
+  let successCount = 0;
+  let failureCount = 0;
+
+  for (const order of openOrders) {
+    try {
+      await submitOrderCancellationById(order.id);
+      successCount++;
+      console.log(`Successfully cancelled order ${order.id}`);
+    } catch (e) {
+      if (e instanceof Error) {
+        failureCount++;
+        console.error(`Failed to cancel order ${order.id}: ${e.message}`);
+      } else {
+        failureCount++;
+        console.error(`Failed to cancel order ${order.id}: An unknown error occurred`);
+      }
+    }
+  }
+
+  console.log(`\nCancellation complete. Successfully cancelled ${successCount} order(s). Failed to cancel ${failureCount} order(s).`);
```

Comment:
Since order submission only submits cancellation requests (doesn't wait to confirm if the order actually got cancelled synchronously), "Cancellation complete" might be misleading (since the cancellations may still be queued to happen, or already have happened, but the db not yet consistent with trading state).

Unless we switch to a completely synchronous flow where we wait on each cancellation, the best we can communication is "this was submitted" (the order could already be cancelled or filled, & the cancellation rejected, etc).